### DISCUSSION TITLE
Can't compile because Magick++.h: No such file or directory

### DISCUSSION THREAD

**Discussion by jackthenewbie** [ImageMagick/ImageMagick#5805] [Help] [answered]
I'm fairly new to installing third party library. so as I was following steps (installing from source):
~/dir/ImageMagick$ `./configure --prefix=/home/pi/dir/project1/ImageMagick LDFLAGS=-L/home/pi/dir/project1/ImageMagick/lib/ && make && make install`
And in the `project1` dir, I have some file call ``test.cpp``
```#include <iostream>
#include "Magick++.h"
using namespace std;
int main()
{
    cout << "Hello World" << endl;
    return 0;
}
```
When compile with ``g++ `Magick++-config --cppflags --cxxflags --ldflags --libs` -fdiagnostics-color=always -g /home/pi/dir/project1/test.cpp -o /home/pi/dir/project1/test`` I has this following error: 
```
Package Magick++-7.Q16HDRI was not found in the pkg-config search path.
Perhaps you should add the directory containing `Magick++-7.Q16HDRI.pc'
to the PKG_CONFIG_PATH environment variable
No package 'Magick++-7.Q16HDRI' found
```
This is my ``Magick++-config --cppflags --cxxflags --ldflags --libs`` command:
```
-fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/home/pi/dir/project1/ImageMagick/include/ImageMagick-7
-fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/home/pi/dir/project1/ImageMagick/include/ImageMagick-7
-L/home/pi/dir/project1/ImageMagick/lib -lMagick++-7.Q16HDRI -lMagickWand-7.Q16HDRI -lMagickCore-7.Q16HDRI
-L/home/pi/dir/project1/ImageMagick/lib -lMagick++-7.Q16HDRI -lMagickWand-7.Q16HDRI -lMagickCore-7.Q16HDRI
```

Here is my ``PKG_CONFIG_PATH``:
```
/home/pi/dir/project1/ImageMagick/lib/pkgconfig
```

Sorry what's part in the [install instruction](https://imagemagick.org/Magick++/Install.html) that's I missed? Can someone help point me there

**Comment by jackthenewbie** [comment, marked as answer]
Problem solved

Edit: As dlemstra recommendation, this is how a solved it.
Apparently, I couldn't remember exactly how I solved it:
1. It could be that when I set PKG_CONFIG_PATH, I didn't restart the machine, so it doesn't apply yet
2. In case of using ``--prefix`` option, you have to set path for ImageMagick lib to ``LD_LIBRARY_PATH=``, so in my case ImageMagick prefix was ``/home/pi/dir/project1/ImageMagick``. Hence I have to do ``export LD_LIBRARY_PATH=/home/pi/dir/project1/ImageMagick/lib``. Then either you give a machine a restart or run ``ldconfig`` (may need sudo to run) to apply config.

Same setting as above, everything should work fine now.

**Reply by dlemstra** [reply]
Would you mind sharing how you solved this in case someone finds your post in the future?
